Author: afa
Date: Wed Jun 16 11:00:27 2010
New Revision: 75419

Modified:
   pypy/trunk/pypy/translator/translator.py
Log:
Don't import the FlowObjSpace too early:
among its dependencies is the rffi module, which needs the compiler to define its types.
This causes problems when using the --cc option.

(the change that broke --cc is r73857, in pyframe.py;
but it feels more robust to defer the import in translator.py)


Modified: pypy/trunk/pypy/translator/translator.py
==============================================================================
--- pypy/translator/translator.py	(original)
+++ pypy/translator/translator.py	Wed Jun 16 11:00:27 2010
@@ -7,7 +7,6 @@
 import autopath, os, sys, types, copy
 
 from pypy.translator import simplify
-from pypy.objspace.flow.objspace import FlowObjSpace
 from pypy.objspace.flow.model import FunctionGraph, checkgraph, Block
 from pypy.tool.ansi_print import ansi_log
 from pypy.tool.sourcetools import nice_repr_for_func
@@ -67,6 +66,7 @@
         else:
             if self.config.translation.verbose:
                 log.start(nice_repr_for_func(func))
+            from pypy.objspace.flow.objspace import FlowObjSpace
             space = FlowObjSpace(self.flowconfig)
             if self.annotator:
                 # ZZZ
Author: afa
Date: Wed Jun 16 11:31:00 2010
New Revision: 75420

Modified:
   pypy/trunk/pypy/translator/goal/targetpreimportedpypy.py
   pypy/trunk/pypy/translator/goal/targetpypystandalone.py
Log:
Remove unneeded import, and really fix --cc=<othercompiler>


Modified: pypy/trunk/pypy/translator/goal/targetpreimportedpypy.py
==============================================================================
--- pypy/translator/goal/targetpreimportedpypy.py	(original)
+++ pypy/translator/goal/targetpreimportedpypy.py	Wed Jun 16 11:31:00 2010
@@ -3,7 +3,6 @@
 import os, sys
 sys.setrecursionlimit(17000)
 
-from pypy.objspace.std.objspace import StdObjSpace
 from pypy.interpreter import gateway
 from pypy.interpreter.error import OperationError
 from pypy.translator.goal.ann_override import PyPyAnnotatorPolicy

Modified: pypy/trunk/pypy/translator/goal/targetpypystandalone.py
==============================================================================
--- pypy/translator/goal/targetpypystandalone.py	(original)
+++ pypy/translator/goal/targetpypystandalone.py	Wed Jun 16 11:31:00 2010
@@ -2,7 +2,6 @@
 
 import os, sys
 
-from pypy.objspace.std.objspace import StdObjSpace
 from pypy.interpreter import gateway
 from pypy.interpreter.error import OperationError
 from pypy.translator.goal.ann_override import PyPyAnnotatorPolicy
